aligent/orocommerce-fees-bundle
# [FEATURE] Implement Payment Processing Fees

Resolves #11 

## Summary
This PR implements Payment Processing Fees for OroCommerce, as well as updating the documentation and installation instructions (as bundle is now on Packagist).

## Changes
* Add Schema Migration to add `processing_fee` to `Order` entity
* Add `AbstractSubtotalFeeProvider` to inject Subtotal Fees into Checkout
* Remove unnecessary `isApplicable` method from `FeeProviderInterface`
* Add `PaymentProcessingFeeProvider` to calculate Processing Fees
* Add `SubtotalFeeProviderInterface`
* Add Form Types for Processing Fee Configuration
* Add System Configuration for Processing Fees
* Add Twig templates for Processing Fee configuration
* Add Translations for Processing Fees (and translate hardcoded strings)
* Update `ConfigurationTest` with new options
* Remove hardcoded Processing Fee label from abstract class
* Add abstract `getFeeLabel` method to abstract class
* Ensure Processing Fees are rounded
* Add `OrderStub` for unit tests
* Add unit tests for `AbstractSubtotalFeeProvider`
* Add unit tests for `PaymentProcessingFeeProvider`
* Update license from deprecated `GPL-3.0` to `GPL-3.0-only`
* Update `README` to reflect current state of Bundle, including updated Instructions and Payment Processing Fee

## Screenshots
**System Configuration**
![2022-07-11_10-44](https://user-images.githubusercontent.com/45612883/178216955-4ccdba73-6885-4282-84b5-aa842238ac32.png)
**Checkout Subtotal**
![2022-07-11_10-46](https://user-images.githubusercontent.com/45612883/178216979-ba6d7128-4e1a-4c9d-bd37-272caf082f88.png)
**Order Confirmation Email**
![2022-07-11_10-46_1](https://user-images.githubusercontent.com/45612883/178217006-b7bb6e76-e095-4d83-8c41-9c7ac3191258.png)
**View Order (Storefront)**
![2022-07-11_10-46_2](https://user-images.githubusercontent.com/45612883/178217027-2b42692a-3f4c-4cc9-9a1e-4fdd59bb51ba.png)
**View Order (Backoffice)**
![2022-07-11_10-47](https://user-images.githubusercontent.com/45612883/178217041-e7b1b5cb-632b-47ea-a0f7-095b3774d5f6.png)
**GET Order (REST API)**
![2022-07-11_10-48](https://user-images.githubusercontent.com/45612883/178217054-d053651f-bb7c-45b3-a4f5-a273b8b13980.png)
